File: inventory-service/src/main/java/com/ecommerce/inventory/entity/Inventory.java
package com.ecommerce.inventory.entity;

import com.ecommerce.shared.enums.ProductColor;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "inventories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Inventory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long productId;
    private String size;

    @Enumerated(EnumType.STRING)
    private ProductColor color;

    private int quantity;
    private double price;

}

File: mobile_app/lib/models/inventory.dart
class Inventory {
  final int id;
  final int productId;
  final String size;
  final String color;
  final double price;
  final int quantity;

  Inventory({
    required this.id,
    required this.productId,
    required this.size,
    required this.color,
    required this.price,
    required this.quantity,
  });

  factory Inventory.fromJson(Map<String, dynamic> json) {
    return Inventory(
      id: json['id'],
      productId: json['productId'],
      size: json['size'],
      color: json['color'],
      price: json['price'],
      quantity: json['quantity'],
    );
  }
}
